Java/java-app/src/main/java/com/example/App.java

```diff
@@ -0,0 +1,70 @@
+package com.example;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOptions;
+import io.javalin.Javalin;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+public class App {
+    private static int counter = 0;
+    private static Firestore db;
+    private static String projectId;
+
+    public static void main(String[] args) {
+        // Initialize Firebase Admin SDK
+        try {
+            GoogleCredentials credentials = GoogleCredentials.getApplicationDefault();
+            FirestoreOptions options = FirestoreOptions.newBuilder().setCredentials(credentials).build();
+            db = options.getService();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        // Create Javalin app
+        Javalin app = Javalin.create().start(7000);
```

Comment:
The Dockerfile correctly specifies port 8080, but it looks like the application is listening to port 7000.